python/ray/data/_internal/logical/operators/join_operator.py

```diff
@@ -217,3 +227,72 @@ def _get_referenced_columns(self, expr: "Expr") -> set[str]:
         visitor = _ColumnReferenceCollector()
         visitor.visit(expr)
         return set(visitor.get_column_refs())
+
+    def _strip_suffix_from_columns(
+        self, columns: set[str], suffix: Optional[str]
+    ) -> set[str]:
+        """Strip a suffix from column names if present.
+
+        Args:
+            columns: Set of column names (potentially with suffixes)
+            suffix: The suffix to strip (e.g., "_l" or "_r")
+
+        Returns:
+            Set of column names with the suffix removed (if it was present)
+        """
+        if suffix is None or suffix == "":
+            return columns
+
+        return {col[: -len(suffix)] if col.endswith(suffix) else col for col in columns}
+
+    def get_column_substitutions(
+        self, side: Optional[JoinSide] = None
+    ) -> dict[str, str]:
+        """Get column substitutions for predicate pushdown.
+
+        This method implements the standard predicate passthrough interface.
+        For Join operators, column substitutions depend on which side is being
+        pushed to, so the `side` parameter must be provided.
```

Comment:
Read this docstring and still felt confused what the term "column substitutions" means. Didn't understand until I read the code that this is referring to adding suffixes to the column names